source/code/scxsystemlib/software/installedsoftwaredepend.cpp

```diff
@@ -425,212 +327,70 @@ Homepage: http://nfs.sourceforge.net/
     */
-    void InstalledSoftwareDependencies::GetRPMQueryResult(int argc, char * argv[], const wstring& tempFileName, std::vector<wstring>& result)
+    void InstalledSoftwareDependencies::GetRPMQueryResult(int argc, char * argv[], std::vector<wstring>& result)
     {
         // We need to redirect stdout to a file so we can capture the result of InvokeRPMQuery() call.
-        
-        // Flush stdout.
-        if (fflush(stdout) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"fflush(stdout) failed", errno), SCXSRCLOCATION);
-        }
-        // Make a copy of original stdout destination.
-        int oldStdOutFD = dup(1);
-        if (oldStdOutFD == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"dup(1) failed", errno), SCXSRCLOCATION);
-        }
-        // Open new stdout destination.
-        int fileFD = open(SCXCoreLib::StrToUTF8(tempFileName).c_str(), O_CREAT | O_RDWR | O_TRUNC, S_IRUSR | S_IWUSR);
-        if (fileFD == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"open(" + tempFileName + L") failed", errno), SCXSRCLOCATION);
-        }
-        // Redirect stdout to the new destination.
-        if (dup2(fileFD, 1) == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"dup2(fileFD, 1) failed", errno), SCXSRCLOCATION);
-        }
-
-        InvokeRPMQuery(argc, argv);
-
-        // Flush new stdout.
-        if (fflush(stdout) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"fflush(stdout) failed", errno), SCXSRCLOCATION);
-        }
-        // Restore stdout to the original destination.
-        if (dup2(oldStdOutFD, 1) == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"dup2(oldStdOutFD, 1) failed", errno), SCXSRCLOCATION);
-        }
-        // Close new stdout destination, not needed anymore.
-        if (close(fileFD) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"close(fileFD) failed", errno), SCXSRCLOCATION);
-        }
-        // Close copy of the original stdout destination.
-        if (close(oldStdOutFD) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"close(oldStdOutFD) failed", errno), SCXSRCLOCATION);
-        }
-
-        SCXCoreLib::SCXHandle<std::wfstream> fs(SCXCoreLib::SCXFile::OpenWFstream(tempFileName, std::ios::in));
-        fs.SetOwner();
-        SCXStream::NLFs nlfs;
-        SCXCoreLib::SCXStream::ReadAllLines(*fs, result, nlfs);
-        fs->close();
+	FILE * fp;
+	
+	const int BUFFER_BLOCK_SIZE = 2048;
+	char contentBlock[BUFFER_BLOCK_SIZE];
+	char * content = NULL;
+	size_t totalBytesRead = 0;
+	size_t bytesRead = 0;
+	size_t currentBytePointer = 0;
+	int errorCode;
+	std::string systemString = "/bin/rpm ";
+	for (int i = 0; i < argc; ++i)
+	{
+	    systemString += std::string(argv[i]) + std::string(" ");
+	}
+	
+	fp = popen(systemString.c_str(), "r");
```

Comment:
Rather than using popen, I think you're better off using SCXProcess::Run(). That gets you the same result, but you get a stream result, which avoids reallocs (and potential memory trashing).

Try that, let me know how it goes.

Be sure to check the size of the error stream - if an error occurs, you probably don't want to ignore it. We often just make sure that the error stream is empty unless we expect output there (even with a success error code).

I'm not quite square on what's happening at 388. You want to check the exit code of the command as well. So typically, we'll return some kind of error if the exit code is non zero OR len(stderr) > 0.